src/main/java/org/galatea/starter/entrypoint/BaseRestController.java

```diff
@@ -4,21 +4,36 @@
 import net.sf.aspect4log.Log;
 import net.sf.aspect4log.Log.Level;
 import org.galatea.starter.utils.Tracer;
-
+import org.owasp.esapi.ESAPI;
 /**
  * Provides base functionality shared by all Fuse REST controllers
  */
 @Slf4j
 @Log(enterLevel = Level.INFO, exitLevel = Level.INFO)
 public abstract class BaseRestController {
+  static final char ENCODER_SEPERATOR = '_';
+
+  /** 
+   * Encodes specified message, to avoid any log forging.
+   */
+  protected String encode(String message){
+
+    message = message.replace( '\n', ENCODER_SEPERATOR ).replace( '\r', ENCODER_SEPERATOR )
+      .replace( '\t', ENCODER_SEPERATOR );
+    
+    message = ESAPI.encoder().encodeForHTML( message );
+
+    return message;
+  }
+
 
   /**
    * Adds the specified requestId to the context for this request (if not null).
    */
   protected void processRequestId(String requestId) {
     if (requestId != null) {
-      log.info("Request received with id: {}", requestId);
-      Tracer.setExternalRequestId(requestId);
+      log.info("Request received with id: {}", encode(requestId));
```

Comment:
Create an _encodedRequestId_ variable and store the result of calling _encode()_ into it so only call the method once.